### ISSUE TITLE
Use Ruby fixtures with expected final state for code action refactor tests

### ISSUE THREAD

**Issue by vinistock** [Shopify/ruby-lsp#531]
Instead of matching the result against and array of text edits, it would be much nicer if we could use a Ruby file as a fixture with the final state after the refactor.

For this, we may need to take the edits returned by running the request, apply them to the document instance and then compare the fixture with `document.source`.

**Comment by andyw8** [comment]
It could possibly use the `diff` format.

**Comment by snutij** [comment]
This is perfect! Thank you for the guidance, I'll go into it, and if needed I open a draft. Feel free to assign it to me 👍

**Comment by vinistock** [comment]
Yes, you would need to put the source inside the result key. So the expectation files would look something like this
```json
{
  "params": [...],
  "result": "
    def method
      abc = "abc"
      foo = 123
      # ...
    end
  " 
}
```
I don't know if we can have multiline strings like this in a JSON. If we can't and need to squish the entire expectation into a single line using `\n`, then I think we need to reconsider since it won't really improve things.

**Comment by snutij** [comment]
This is not possible to have multiline in JSON. For my previous example `extraction_with_empty_line_between` this will lead to this expected result:

```
"params": [foo],
"result":  "def method\n  abc = \"abc\"\n\n  foo = 123\n\n \n  new_variable = { some: \"hash\" } hash = {\n    info: new_variable,\n  }\nend\n"
``` 

In my opinion (someone unfamiliar with LSP specification) is still more readable than an array of text edits. But not sure it's worth implementing it and having a pattern not similar to other requests for this specific request.   

I let you decide, both are OK to me 👍

**Comment by vinistock** [comment]
Yeah, I agree with you that it's not worth making this request different than others for now. We may want to re-think our test framework in a way that better handles parameters, but I think it's not worth the investment immediately.

Thank you for looking into it 🙏.
